CleverSDK/Classes/CLVOAuthManager.m

```diff
@@ -93,8 +117,52 @@ + (NSString *)state {
 }
 
 + (void)login {
+    NSString *state = [self generateRandomString:32];
+    [self setState:state];
+    
+    NSString *safariURLString = [NSString stringWithFormat:@"https://clever.com/oauth/authorize?response_type=code&client_id=%@&redirect_uri=%@&state=%@",
+                                 [self clientId], [self redirectUri], [self state]];
+    
+    NSString *cleverAppURLString = [NSString stringWithFormat:@"com.clever://oauth/authorize?response_type=code&client_id=%@&redirect_uri=%@&state=%@&sdk_version=%@", [self clientId], [self redirectUri], [self state], SDK_VERSION];
+    
     CLVOAuthManager *manager = [self sharedManager];
-    [[manager clvLogin] login];
+    if (manager.districtId) {
+        NSString *districtId = manager.districtId;
+        if (districtId == nil) {
+            districtId = @"";
+        }
+        safariURLString = [NSString stringWithFormat:@"%@&district_id=%@", safariURLString, districtId];
+        cleverAppURLString = [NSString stringWithFormat:@"%@&district_id=%@", safariURLString, districtId];
+    }
+    
+    // iOS 8 - always use Safari. Clever App not supported
+    if (SYSTEM_VERSION_LESS_THAN(@"9.0")) {
+        [[UIApplication sharedApplication] openURL: [NSURL URLWithString:safariURLString]];
+        return;
+    }
+    
+    // Switch to native Clever app if possible
+    if ([[UIApplication sharedApplication] canOpenURL:[NSURL URLWithString:cleverAppURLString]]) {
+        [[UIApplication sharedApplication] openURL:[NSURL URLWithString:cleverAppURLString] options:@{} completionHandler:nil];
```

Comment:
It looks like `openURL` was deprecated in iOS 10 in favor of `open`. Not sure why adding the extra parameters allows this to complete without the delay...I'm guessing it is now a wrapper for the `open` command, but we shouldn't rely on that. Unfortunate that we don't have an iOS 9 device but we should probably special-case that version.

Reply:
j/k it looks like I was looking at the Swift docs. Either way, we should still condition on version